sherpa/plot/pylab_backend.py

```diff
@@ -19,853 +19,803 @@
-                  barsabove=barsabove,
-                  zorder=zorder)
-
-
-_linestyle_map = {
-    'noline': ' ',
-    'solid': '-',
-    'dot': ':',
-    'dash': '--',
-    'dotdash': '-.',
-}
-
-
-def _set_line(line, linecolor=None, linestyle=None, linewidth=None):
-    """Apply the line attributes, if set.
-
-    Parameters
-    ----------
-    line : matplotlib.lines.Line2D
-        The line to change
-    linecolor, linestyle, linewidth : optional
-        The attribute value or None.
-
-    """
-
-    def set(label, newval):
-        func = getattr(line, 'set_' + label)
-        func(newval)
-
-    if linecolor is not None:
-        set('color', linecolor)
-
-    if linestyle is not None:
-        set('linestyle', _linestyle_map[linestyle])
+            pass
 
-    if linewidth is not None:
-        set('linewidth', linewidth)
+        axes = plt.gca()
+        zorder = self.find_zorder(axes)
 
+        # Do not draw a line connecting the points
+        #
+        # Unlike plot, using errorbar for both cases.
+        #
 
-# There is no support for alpha in the Plot.vline class
-#
-def vline(x, ymin=0, ymax=1,
-          linecolor=None,
-          linestyle=None,
-          linewidth=None,
-          overplot=False, clearwindow=True):
-
-    axes = setup_axes(overplot, clearwindow)
-
-    line = axes.axvline(x, ymin, ymax)
-    _set_line(line, linecolor=linecolor, linestyle=linestyle,
-              linewidth=linewidth)
-
+        axes.errorbar(xmid, y, yerr, xerr,
+                      color=color,
+                      alpha=alpha,
+                      linestyle='',
+                      marker=marker,
+                      markersize=markersize,
+                      markerfacecolor=markerfacecolor,
+                      ecolor=ecolor,
+                      capsize=capsize,
+                      barsabove=barsabove,
+                      linewidth=linewidth,
+                      label=label,
+                      zorder=zorder)
+
+    # Note that this is an internal method that is not wrapped in
+    # @ translate_args
+    # This is called from methods like plot, histo, etc.
+    # so the arguments passed to this method are already translated and we do
+    # not want to apply the translation a second time.
+    def _set_line(self, line, linecolor=None, linestyle=None, linewidth=None):
+        """Apply the line attributes, if set.
+
+        Parameters
+        ----------
+        line : matplotlib.lines.Line2D
+            The line to change
+        linecolor, linestyle, linewidth : optional
+            The attribute value or None.
+
+        """
+
+        def set(label, newval):
+            func = getattr(line, 'set_' + label)
+            func(newval)
+
+        if linecolor is not None:
+            set('color', linecolor)
+
+        if linestyle is not None:
+            set('linestyle', linestyle)
+
+        if linewidth is not None:
+            set('linewidth', linewidth)
```

Comment:
We could change this to

```suggestion
        def set(label, newval):
            if newval is None:
                return
            func = getattr(line, 'set_' + label)
            func(newval)

        set('color', linecolor)
        set('linestyle', linestyle)
        set('linewidth', linewidth)
```

but is it worth it?